src/securityPolicy.c

```diff
@@ -276,9 +276,11 @@ security_policy_t policyForSignTxInit(
 		ASSERT(false);
 	}
 
-	WARN_IF(!is_tx_network_verifiable(numOutputs, numWithdrawals, txSigningMode));
+	DENY_UNLESS(isValidNetworkId(networkId));
 
-	WARN_IF(networkId != MAINNET_NETWORK_ID);
+	WARN_UNLESS(is_tx_network_verifiable(numOutputs, numWithdrawals, txSigningMode));
```

Comment:
We aimed to get rid of `is_tx_network_verifiable` --- if tx body contains network id explicitly, we don't care whether it is included in ouputs or wihdrawals.

Also, the warning
```
			// technically, no withdrawals/pool reg. certificate as well, but the UI message would be too long
			ui_displayPaginatedText(
			        header,
			        "no outputs, cannot verify network id",
			        this_fn
			);

```
in signTx.c is no longer needed --- that's the "related logic" I mentioned before.

Reply:
Ah yes, since we always know what the networkId is, or at least code it into the tx. Ok, done!